src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+
+    def start(self):
+        reduced_events = dict()
+        for sel in self.particles:
+            # TODO: not very elegant to pass them this way, refactor later
+            if sel.epp.quality_criteria != self.opt_result.precuts.quality_criteria:
+                self.log.warning(
+                    "Precuts are different from precuts used for calculating "
+                    "g/h / theta cuts. Provided precuts:\n%s. "
+                    "\nUsing the same precuts as g/h / theta cuts:\n%s. "
+                    % (
+                        sel.epp.to_table(functions=True)["criteria", "func"],
+                        self.opt_result.precuts.to_table(functions=True)[
+                            "criteria", "func"
+                        ],
+                    )
+                )
+                sel.epp = EventPreProcessor(
+                    parent=sel,
+                    quality_criteria=self.opt_result.precuts.quality_criteria,
+                )
+
+            if sel.epp.gammaness_classifier != self.opt_result.gh_cuts.meta["CLFNAME"]:
+                self.log.warning(
+                    "G/H cuts are only valid for gammaness scores predicted by "
+                    "the same classifier model. Requested model: %s. "
+                    "Model used, so that g/h cuts are valid: %s."
+                    % (
+                        sel.epp.gammaness_classifier,
+                        self.opt_result.gh_cuts.meta["CLFNAME"],
+                    )
+                )
+                sel.epp.gammaness_classifier = self.opt_result.gh_cuts.meta["CLFNAME"]
+
+            self.log.debug("%s Precuts: %s" % (sel.kind, sel.epp.quality_criteria))
+            evs, cnt, meta = sel.load_preselected_events(self.chunk_size, self.obs_time)
+            # Only calculate event weights if background or sensitivity should be calculated.
+            if self.do_background:
+                # Sensitivity is only calculated, if do_background and do_benchmarks is true.
+                if self.do_benchmarks:
+                    evs = sel.make_event_weights(
+                        evs, meta["spectrum"], self.sens.fov_offset_bins
+                    )
+                # If only background should be calculated,
+                # only calculate weights for protons and electrons.
+                elif sel.kind in ("protons", "electrons"):
+                    evs = sel.make_event_weights(evs, meta["spectrum"])
+
+            reduced_events[sel.kind] = evs
+            reduced_events[f"{sel.kind}_count"] = cnt
+            reduced_events[f"{sel.kind}_meta"] = meta
+            self.log.debug(
+                "Loaded %d %s events" % (reduced_events[f"{sel.kind}_count"], sel.kind)
+            )
+            if sel.kind == "gammas":
+                self.signal_is_point_like = (
+                    meta["sim_info"].viewcone_max - meta["sim_info"].viewcone_min
+                ).value == 0
+
+        if self.signal_is_point_like:
```

Comment:
I think this should also be a plain error: the configuration should match the passed data.